frontend/src/employee-frontend/components/holiday-term-periods/TermPeriodEditor.tsx

```diff
@@ -23,11 +26,20 @@ export default React.memo(function TermPeriodEditor() {
     [navigate]
   )
 
+  const findPreschoolTermById = (
+    id: string,
+    allPreschoolTerms: PreschoolTerm[]
+  ) =>
+    termId !== 'new'
+      ? allPreschoolTerms.find((term) => term.id === termId)
+      : undefined
```

Comment:
Id parameter is not used. 

If you want to get fancy and memoize the result so that the list won't be searched on every render (not that it takes a lot of power), that could also be done like this:

```
  const term: Result<PreschoolTerm | undefined> = useMemo(() => {
    if (termId === 'new') return Success.of(undefined)

    return preschoolTerms
      .map((allTerms) => allTerms.find((t) => t.id === termId))
      .chain((t) =>
        t ? Success.of(t) : Failure.of({ message: 'Term not found' })
      )
  }, [termId, preschoolTerms])

  return (
    <Container>
      <ContentArea opaque>
        {renderResult(
          combine(preschoolTerms, term),
          ([preschoolTerms, term]) => (
            <PreschoolTermForm
              term={term}
              allTerms={preschoolTerms}
```

This would also handle the error case slightly better where the term is not found from the list by displaying an error instead of passing `undefined` to `PreschoolTermForm` and implicitly switching from edit mode to create mode.

Reply:
Good points, I replaced the block with the snippet you provided, kudos!